dot/network/discovery.go

```diff
@@ -208,16 +210,8 @@ func (d *discovery) findPeers(ctx context.Context) {
 
 			logger.Trace("found new peer via DHT", "peer", peer.ID)
 
-			// found a peer, try to connect if we need more peers
-			if len(d.h.Network().Peers()) < d.maxPeers {
```

Comment:
why is this removed?

Reply:
it seems `AddToPeerSet` doesn't actually connect to the peer

Reply:
`maxPeers ` and `minPeers` are maintained in the peerset.

Reply:
I can't find where in the code it actually connects to the peer, it seems all `connect` calls are removed in favour of `AddToPeerSet` which doesn't seem to connect (also integration tests are failing) - I think you probably need to leave the `connect` calls in the network package, as the peerset shouldn't deal with the actual networking, just tracking peer info